docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
```

Comment:
this rule is based on the tslint rule (which includes only the literals), but yeah, we can broaden the scope by including the callback too. I think we should consider renaming the rule, though...

thoughts?